### PR TITLE
new click-system + rules/effects

### LINKED ISSUE #228: Disease  / Sickness / Nerf / Curse & Blessing Ability Management
**Is your feature request related to a problem? Please describe.**
Section 196 suffering from GHOULBITE
#230 
Choose the ability that gets nerfed till a cure is found

Section 197
Non choosable, can be used up
```

**Describe the solution you'd like**
check for curses and what ability is affected how much (kind of like Buffs from inventory or Blessings)

### LINKED ISSUE #552: Make interactive Java sections less verbose
the whole method looks perfect... but so verbose. now I feel bad for this design :D 
but I don't have a real idea to make it shorter/more declarative...

_Originally posted by @keykey7 in https://github.com/keykey7/fabbwled/pull/548#discussion_r1399349965_

### PR DESCRIPTION
**Pull request by keykey7**
### What?
Easier interface to write sections in Java.
Methods to write game-rules and effects (like COMBAT+1) standalone.

### Why?
- Writing Java-Sections was tedious work mainly because of the strict separation of view and controller.
- Added a better ground floor for random effects that were not started until now.

### How?
- Fluent interface of SectionNode was updated (among others), to include a `ContainerNode clickable(PlayerChange action, ...)`, where `PlayerChange` is a lambda that is executed on-click.
- new interface `OnAttributeRefresh`: every controller implementing it will get a chance to update the `PlayerDto` before it is used. So any attribute can be changed (not just DEFENCE and similar obvious things).
- `PlayerDto` was extended by poison, curses, diseases. mostRecentDiceRoll was moved here from it's own service (this way it becomes part of the state).

### Anything Else?
Implements a hardcore and an easy section to illustrate the usage of the if-else structure.

fixes #552 
fixes #228 
fixes https://github.com/keykey7/fabbwled/issues/475
fixes https://github.com/keykey7/fabbwled/issues/521

Apologies due to the size of the PR.

### CODE REVIEW

**Comment by keykey7** [comment]
ability rolls are possible in non-yaml: https://github.com/keykey7/fabbwled/issues/229

**Review by tamer545 — APPROVED** [review]
This makes section creation much easier, nice

**Review by Lion-alt — APPROVED** [review]
Looks good. Place where shards are stored changes here @Noeell .

File: fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/book/yaml/YamlSectionHandler.java [hunk 1]
```diff
@@ -3,27 +3,22 @@
...
 
-import java.util.ArrayList;
-import java.util.function.Function;
-
 @RequiredArgsConstructor
 public class YamlSectionHandler implements SectionHandler {
     private final PlayerSession playerSession;
     private final YamlSection section;
 
-    private ArrayList<Function<PlayerSession.PlayerDto, PlayerSession.PlayerDto>> handlers = new ArrayList<>();
```
Comment (Nilstrieb) [inline fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/book/yaml/YamlSectionHandler.java:18]: In #556, I'm using the handlers for more fun things: https://github.com/keykey7/fabbwled/pull/556/files#diff-5e3224d8e0cd08c5df61abd75b431da8062e4c0cdd75c8a311a4d4eb90253890R93
That's gonna conflict, what should we do with those?

**Review by adrsch04 — APPROVED** [review]
👍

**Review by kimberlymo — APPROVED** [review]
LGTM 👍 will for sure help us with implementing new features

**Review by Desi7 — APPROVED** [review]
Looking good 👍 very helpfull for writing sections

File: fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/book/yaml/YamlSectionHandler.java (hunk 1 above)
Comment (keykey7) [reply fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/book/yaml/YamlSectionHandler.java:18]: merged

**Pull request merged**